exporter/loadbalancingexporter/log_exporter_test.go

```diff
@@ -287,6 +287,58 @@ func TestLogsWithoutTraceID(t *testing.T) {
 	assert.Len(t, sink.AllLogs(), 1)
 }
 
+// this test validates that exporter is can concurrently change the endpoints while consuming logs.
+func TestConsumeLogs_ConcurrentResolverChange(t *testing.T) {
```

Comment:
This test passes on main as well, it might not trigger the error condition this PR is aiming to fix.

Reply:
Not sure how it passed for you. Here is the draft PR with the tests only https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31566